### PR TITLE
chore(deps): upgrade to 1.49 toolchain

### PR DESCRIPTION
**Pull request by fanatid**
https://github.com/rust-lang/rust/releases/tag/1.49.0
https://blog.rust-lang.org/2020/12/31/Rust-1.49.0.html

### CODE REVIEW

File: lib/remap-lang/src/expression/function.rs [hunk 1]
```diff
@@ -111,10 +111,9 @@ impl Function {
             .enumerate()
             .filter(|(_, p)| p.required)
             .filter(|(_, p)| !list.keywords().contains(&p.keyword))
-            .map(|(i, p)| {
+            .try_for_each(|(i, p)| -> Result<_> {
```
Comment (Hoverbear) [inline lib/remap-lang/src/expression/function.rs:114]: Love this.

File: src/sinks/new_relic_logs.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ use serde::{Deserialize, Serialize};
 use snafu::Snafu;
 
 // New Relic Logs API accepts payloads up to 1MB (10^6 bytes)
-const MAX_PAYLOAD_SIZE: usize = 1_000_000 as usize;
+const MAX_PAYLOAD_SIZE: usize = 1_000_000usize;
```
Comment (Hoverbear) [inline src/sinks/new_relic_logs.rs:17]:
```suggestion
const MAX_PAYLOAD_SIZE: usize = 1_000_000_usize;
```

**Review by Hoverbear — APPROVED** [review]

File: lib/remap-lang/src/expression/function.rs (hunk 1 above)
Comment (fanatid) [reply lib/remap-lang/src/expression/function.rs:114]: :100:

File: src/config/loading.rs [hunk 3]
```diff
@@ -22,8 +22,8 @@ lazy_static! {
 
 /// Merge the paths coming from different cli flags with different formats into
 /// a unified list of paths with formats.
-pub fn merge_path_lists<'a>(
-    path_lists: Vec<(&'a [PathBuf], FormatHint)>,
+pub fn merge_path_lists(
+    path_lists: Vec<(&'_ [PathBuf], FormatHint)>,
```
Comment (pablosichert) [inline src/config/loading.rs:26]: Is the explicitly elided lifetime necessary? I assume `&[PathBuf]` would be sufficient instead of `&'_ [PathBuf]`, unless I'm missing something.

The clippy rule https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes says

> Lifetime bounds such as `impl Foo + 'a` and `T: 'a` must be elided with the placeholder notation `'_` because the fully elided notation leaves the type bound to `'static`.

which doesn't seem to be the case here.

Comment (fanatid) [reply src/config/loading.rs]: They are not required, thanks! Removed.

Comment (binarylogic) [reply src/config/loading.rs]: Is this a clippy rule we should enable?

Comment (fanatid) [reply src/config/loading.rs]: This rule (`needless_lifetimes`) already enabled (`warn` level) but do not report about `&'_ x` because we do not define lifetime explicitly (like `<'a>`)

**Review by JeanMertz — APPROVED** [review]

File: src/kubernetes/state/instrumenting.rs [hunk 4]
```diff
@@ -136,19 +136,16 @@ mod tests {
...
     /// This is required because we assert on a global state, and we don't
     /// want interference.
-    fn tests_lock() -> MutexGuard<'static, ()> {
+    async fn tests_lock() -> MutexGuard<'static, ()> {
         static INSTANCE: OnceCell<Mutex<()>> = OnceCell::new();
-        INSTANCE
-            .get_or_init(|| Mutex::new(()))
-            .lock()
-            .unwrap_or_else(|err| err.into_inner())
+        INSTANCE.get_or_init(|| Mutex::new(())).lock().await
```
Comment (MOZGIII) [inline src/kubernetes/state/instrumenting.rs:141]: Looks fine, but why is this required?

Comment (fanatid) [reply src/kubernetes/state/instrumenting.rs:141]: Clippy rule [await_holding_lock](https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock):
```
error: this MutexGuard is held across an 'await' point. Consider using an async-aware Mutex type or ensuring the MutexGuard is dropped before calling await.
   --> src/kubernetes/state/instrumenting.rs:151:13
    |
151 |         let _guard = tests_lock();
    |             ^^^^^^
    |
note: these are all the await points this lock is held through
   --> src/kubernetes/state/instrumenting.rs:151:9
    |
151 | /         let _guard = tests_lock();
152 | |
153 | |         let (mut writer, mut events_rx, mut actions_tx) = prepare_test();
154 | |
...   |
175 | |         join.await.unwrap();
176 | |     }
    | |_____^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock
```
